app/models/custom_value/calculated_value_strategy.rb

```diff
@@ -38,7 +38,24 @@ def typed_value
   end
 
   def formatted_value
-    integer_value? ? value.to_s : number_with_delimiter(value.to_s)
+    return "" if value.blank?
+
+    if integer_value?
+      number_with_delimiter(value.to_i)
+    else
+      formatted = number_with_precision(value.to_f,
+                                        precision: 3,
+                                        strip_insignificant_zeros: true,
+                                        delimiter: I18n.t("number.format.delimiter"),
+                                        separator: I18n.t("number.format.separator"))
+
+      # Ensure at least one decimal place for floats
+      if formatted.exclude?(I18n.t("number.format.separator"))
+        "#{formatted}#{I18n.t('number.format.separator')}0"
+      else
+        formatted
+      end
```

Comment:
I wanted to suggest not using `strip_insignificant_zeros`, but it doesn't look like our own stripping logic will be more compact, so only suggesting to extract translations:
```suggestion
      delimiter = I18n.t("number.format.delimiter")
      separator = I18n.t("number.format.separator"))
      formatted = number_with_precision(value.to_f,
                                        precision: 3,
                                        strip_insignificant_zeros: true,
                                        delimiter:,
                                        separator:)

      # Ensure at least one decimal place for floats
      if formatted.exclude?(separator)
        "#{formatted}#{separator}0"
      else
        formatted
      end
```